packages/labs/src/components/timezone-select/timezoneSelect.tsx

```diff
@@ -0,0 +1,509 @@
+                return timezone1 < timezone2 ? -1 : 1;
+            }
+            // Descending order of population
+            return population2 - population1;
+        }
+        // Ascending order of offset
+        return offset1 - offset2;
+    });
+
+    // Choose the most populous locations for each offset
+    const initialTimezones: ITimezoneItem[] = [];
+    let prevOffset: number;
+    for (const timezone of timezones) {
+        const curOffset = timezoneToMetadata[timezone].offset;
+        if (prevOffset === undefined || prevOffset !== curOffset) {
+            initialTimezones.push(toTimezoneItem(timezone, date));
+            prevOffset = curOffset;
+        }
+    }
+    return initialTimezones;
+}
+
+function getLocalTimezoneItem(date: Date): ITimezoneItem | undefined {
+    const timezone = getLocalTimezone();
+    if (timezone) {
+        const timestamp = date.getTime();
+        const zonedDate = moment.tz(timestamp, timezone);
+        const offsetAsString = zonedDate.format("Z");
+        return {
+            iconName: "locate",
+            key: `${timezone}-local`,
+            label: offsetAsString,
+            text: "Current timezone",
+            timezone,
+        };
+    } else {
+        return undefined;
+    }
+}
+
+/**
+ * Get the local timezone.
+ * Note that we are not guaranteed to get the correct timezone in all browsers,
+ * so this is a best guess.
+ * https://momentjs.com/timezone/docs/#/using-timezones/guessing-user-timezone/
+ */
+function getLocalTimezone(): string {
+    return moment.tz.guess();
+}
+
+function getTimezoneQueryCandidates(timezones: ITimezoneItem[], date: Date): { [timezone: string]: string[] } {
+    const timezoneToQueryCandidates: { [timezone: string]: string[] } = {};
+
+    for (const { timezone } of timezones) {
+        const { abbreviation, offsetAsString } = getTimezoneMetadata(timezone, date);
+        const candidates = [
+            timezone,
+            abbreviation,
+            offsetAsString,
+
+            // Split timezone string.
+            // For example, "America/Los_Angeles" -> "America Los Angeles"
+            timezone.split(/[/_]/).join(" "),
+
+            // Don't require leading offset 0.
+            // For example, "-07:00" -> "-7:00"
+            offsetAsString.replace(/^([+-])0(\d:\d{2})$/, "$1$2"),
+        ].map(normalizeText);
+
+        timezoneToQueryCandidates[timezone] = timezoneToQueryCandidates[timezone] || [];
+        timezoneToQueryCandidates[timezone].push(...candidates);
+    }
+
+    return timezoneToQueryCandidates;
+}
+
+function toTimezoneItem(timezone: string, date: Date): ITimezoneItem {
+    const { abbreviation, offsetAsString } = getTimezoneMetadata(timezone, date);
+    return {
+        iconName: undefined,
+        key: timezone,
+        label: offsetAsString,
+        text: timezone + (abbreviation ? ` (${abbreviation})` : ""),
+        timezone,
+    };
+}
+
+interface ITimezoneMetadata {
```

Comment:
same here, I would put this in another file, `timezoneSelectTypes.ts` or `timezoneSelectInterfaces.ts`